includes/forms/class-llms-form-field.php

```diff
@@ -846,9 +874,8 @@ protected function prepare_value() {
 		}
 
 		// Auto-populate field from the datastore if we have a user and datastore information.
-		if ( is_null( $user_val ) && is_user_logged_in() && $this->settings['data_store_key'] && in_array( $this->settings['data_store'], array( 'users', 'usermeta' ), true ) ) {
-			$user     = wp_get_current_user();
-			$user_val = $user->get( $this->settings['data_store_key'] );
+		if ( is_null( $user_val ) && ( ! is_null( $this->user ) ) && $this->settings['data_store_key'] && in_array( $this->settings['data_store'], array( 'users', 'usermeta' ), true ) ) {
```

Comment:
I think the logic can be simplified here. In `define_data_source()` we validate the presence of a data store so here I think all we need to do is verify that a `data_store_key` exists for the field, right?

```suggestion
		if ( is_null( $user_val ) && ( ! is_null( $this->user ) ) && $this->settings['data_store_key'] ) {
```

Reply:
What you say here makes totally sense, if `$this->user` stays as I said above :P